src/asar-utils.ts

```diff
@@ -1,14 +1,41 @@
 import * as asar from 'asar';
+import { execFileSync } from 'child_process';
 import * as crypto from 'crypto';
 import * as fs from 'fs-extra';
 import * as path from 'path';
+import * as minimatch from 'minimatch';
+import * as os from 'os';
 import { d } from './debug';
 
 export enum AsarMode {
   NO_ASAR,
   HAS_ASAR,
 }
 
+export type FuseASARsOptions = {
+  x64: string;
+  arm64: string;
+  output: string;
+
+  lipo?: string;
+  uniqueAllowList?: string;
+};
+
+// See: https://github.com/apple-opensource-mirror/llvmCore/blob/0c60489d96c87140db9a6a14c6e82b15f5e5d252/include/llvm/Object/MachOFormat.h#L108-L112
+const MACHO_MAGIC = new Set([
+  // 32-bit Mach-O
+  0xfeedface,
+  0xcefaedfe,
+
+  // 64-bit Mach-O
+  0xfeedfacf,
+  0xcffaedfe,
+
+  // Universal
+  0xcafebabe,
```

Comment:
I believe there's a slight risk of misidentifying Java bytecode files as macho, but in that case `lipo` will fail anyway so maybe we shouldn't bother?